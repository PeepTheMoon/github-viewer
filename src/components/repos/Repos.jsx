import React from 'react';
import PropTypes from 'prop-types';

const Repos = ({ repos }) => {
  const repoList = repos.map(repo => (
    <li key={repo}>{repo}</li>
  ));

  return (
    <ul>
      User Repos:
      {repoList}
    </ul>
  );
};

Repos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Repos;
