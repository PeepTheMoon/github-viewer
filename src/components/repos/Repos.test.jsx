import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos';

describe('Repos component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Repos repos={['https://github.com/peepthemoon/repos']} />);
    expect(wrapper).toMatchSnapshot();
  });
});
