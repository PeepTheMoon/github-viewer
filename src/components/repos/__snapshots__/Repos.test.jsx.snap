// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders App 1`] = `
<ul>
  User Repos:
  <li
    key="https://github.com/peepthemoon/repos"
  >
    https://github.com/peepthemoon/repos
  </li>
</ul>
`;
